.github/workflows/test-requirements.yml

```diff
@@ -1,71 +0,0 @@
-name: Validate requirements
```

Comment:
This file appears to be identical to https://github.com/mlflow/mlflow/blob/master/.github/workflows/requirements.yml, except for the title. Not sure why we need / how we ended up with both. Removing for now.